addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -276,7 +277,11 @@ export class SearchBar extends Component {
 
         const options = await this.orm.call(relation, "name_search", [], {
             args: domain,
-            context: { ...this.env.searchModel.globalContext, ...field.context },
+            context: {
+                ...this.env.searchModel.globalContext,
+                ...field.context,
+                ...makeContext([searchItem.context]),
```

Comment:
Hello. searchItem.context can contain variables like self and in that case makeContext crashes  (as attested by the failing test). Normally, field contexts are prepared in _getSearchItemContext but here we do not have an autocompleteValue. I propose to simply prepare a searchItemContext similarly to what is done a bit above for searchItem.domain.